textures/metro_control_panel_001_pegboard.py

```diff
@@ -1,8 +1,15 @@
 """A collection of stuff related to https://stalburg.net/Body_message#Pegs."""
+from functools import reduce
+from typing import Callable, Iterable, NamedTuple, Sequence, Tuple, TypeVar, Collection
 
-from typing import Tuple
+T = TypeVar("T")
 
+NorMask = Callable[[Sequence[bool], Sequence[bool]], Tuple[bool]]
 Pegs = Tuple[Tuple[str, str], ...]
+PegGroup = NamedTuple("PegGroup", [("name", str), ("pegs", Sequence[bool]), ("markers", Sequence[bool])])
+
+# █◉█oo█◉█◉█ooo█◉█◉█◉ooo█◉ | █◉█◉█◉oooo█◉█◉oooo█◉█◉█◉ | █◉█◉█◉█◉oooooooooooooooo
```

Comment:
please avoid using unicode characters - I know it's tempting, but it makes things overly complicated. One has to use a unicode lookup thingie to find the character and then paste it here. Doing the reverse, i.e. finding the unicode code point for a unicode character, is even harder. If you want to use unicode, please add constants and unicode code point escape sequences.